### PR TITLE
feat(ui5-side-navigation): make items customizable

### LINKED ISSUE #7808: SideNavigationItem Font Weight Issue in Horizon Theme
### Bug Description

SideNavigationItem which does not have SideNavigationSubItem - still coming as "bold" in Horizon theme.
For not-so-important items which should not be highlighted... they are part of fixed items.

### Affected Component

SideNavigationItem

### Expected Behaviour

We want the control to manually set "not bold" for some SideNavigationItem based on requirement.

### Isolated Example

_No response_

### Steps to Reproduce

1.
2.
3.
...

### Log Output, Stack Trace or Screenshots

Screenshot 2023-11-07 at 11 54 27 AM

### Priority

Medium

### UI5 Web Components Version

1.22.0

### Browser

Chrome

### Operating System

_No response_

### Additional Context

_No response_

### Organization

CBC

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

### PR DESCRIPTION
**Pull request by TeodorTaushanov**
Texts styles are set to the components, so they can be customized.

fixes #7808

### CODE REVIEW

File: packages/fiori/src/themes/SideNavigationItemBase.css [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+	border-width: var(--_ui5_side_navigation_group_border_width);
+}
+
+.ui5-sn-item-group[aria-expanded="true"]::before,
+.ui5-sn-item-level1[aria-expanded="true"]::before {
+	border-width: var(--_ui5_side_navigation_group_expanded_border_width);
+}
+
+/* borders are drawn using a pseudo element.
+in some themes, the border is drawn only on the bottom.
```
Comment (BorisDafov) [inline packages/fiori/src/themes/SideNavigationItemBase.css:61]: better:
at the bottom

File: packages/fiori/src/themes/SideNavigationItemBase.css [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+in high contrast themes the border is on all sides and there is extra border on hover */
```
Comment (BorisDafov) [inline packages/fiori/src/themes/SideNavigationItemBase.css:62]: I suggest:
In high contrast themes, the border is present on all sides, and there is an additional border that appears on hover.

**Review by BorisDafov — CHANGES_REQUESTED** [review]
minor

**Review by alexandar-mitsev — CHANGES_REQUESTED** [review]
in the current implementation it is not possible to override the item in the overflow, maybe you can add a part to allow this?

**Review by alexandar-mitsev — APPROVED** [review]

**Pull request merged**